controller/instance_manager_controller.go

```diff
@@ -334,6 +334,21 @@ func (imc *InstanceManagerController) syncInstanceManager(key string) (err error
 		}
 	}()
 
+	// An instance manager pod for v2 volume need to consume huge pages, and disks managed by the
+	// pod is unable to managed by another pod. Therefore, if an instance manager pod is running on a node,
+	// an extra instance manager pod for v2 volume should not be created.
+	if types.IsDataEngineV2(im.Spec.DataEngine) {
+		if im.Status.CurrentState == longhorn.InstanceManagerStateStopped {
```

Comment:
Is there a reason the state is being checked both here and again in `canProceedWithInstanceManagerSync`?

Reply:
Refactored the codes. Can you help review it again? Thank you.